soroban-env-host/src/expiration_ledger_bumps.rs

```diff
@@ -9,4 +9,22 @@ pub struct LedgerBump {
 }
 
 #[derive(Clone, Default)]
-pub struct ExpirationLedgerBumps(pub Vec<LedgerBump>);
+pub struct ExpirationLedgerBumps(Vec<LedgerBump>);
+
+impl ExpirationLedgerBumps {
+    pub(crate) fn metered_push(
+        &mut self,
+        b: impl AsBudget,
+        elem: LedgerBump,
+    ) -> Result<(), HostError> {
+        // The worst case cost of a `vec.push` requires heap reallocation and copying of old data.
+        // Thus we use the cost of instantiating a size=1 `Vec` as the estimate of the amortized
+        // cost for it.
+        metered_clone::charge_container_bulk_init_with_elts::<Vec<LedgerBump>, LedgerBump>(
```

Comment:
Nice, I believe that's more fair and efficient than just using immutable metered vec without any consideration.